File: src/test/functional/org/apache/prepbuddy/functionaltests/specs/TypeInferenceTestSpec.java
package org.apache.prepbuddy.functionaltests.specs;

import org.apache.prepbuddy.functionaltests.framework.DatasetAssertions;
import org.apache.prepbuddy.functionaltests.framework.DatasetTestSpec;
import org.apache.prepbuddy.functionaltests.framework.TestableDataset;
import org.apache.prepbuddy.qualityanalyzers.DataType;
import org.apache.prepbuddy.rdds.AnalyzableRDD;
import org.apache.spark.api.java.JavaRDD;

public class TypeInferenceTestSpec extends DatasetTestSpec {

    public TypeInferenceTestSpec(TestableDataset dataset) {
        super(dataset);
    }

    @Override
    public void executeTest(JavaRDD<String> testableRDD) {
        AnalyzableRDD analyzableRDD = new AnalyzableRDD(testableRDD);
        analyzableRDD.cache();
        DataType actualDataType = analyzableRDD.inferType(0);
        DatasetAssertions.assertType(DataType.MOBILE_NUMBER, actualDataType);
    }
}

File: src/test/functionaltest/specs/TypeInferenceTestSpec.scala
package specs

import framework.{DatasetAssertion, DatasetTestResults, DatasetTestSpec, TestableDataset}
import org.apache.datacommons.prepbuddy.qualityanalyzers.{DataType, MOBILE_NUMBER}
import org.apache.datacommons.prepbuddy.rdds.TransformableRDD
import org.apache.spark.rdd.RDD

class TypeInferenceTestSpec(dataset: TestableDataset) extends DatasetTestSpec(dataset) {

    override protected var testResults: DatasetTestResults = _

    override def executeTest(testableRDD: RDD[String]): Unit = {
        val transformableRDD: TransformableRDD = new TransformableRDD(testableRDD)
        transformableRDD.cache()
        val actualDatatype: DataType = transformableRDD.inferType(0)
        DatasetAssertion.assertType(MOBILE_NUMBER, actualDatatype)
    }
}
